src/main/java/uk/gov/companieshouse/api/testdata/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,101 @@
+package uk.gov.companieshouse.api.testdata.service.impl;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.testdata.Application;
+import uk.gov.companieshouse.api.testdata.exception.DataException;
+import uk.gov.companieshouse.api.testdata.model.entity.Roles;
+import uk.gov.companieshouse.api.testdata.model.entity.Users;
+import uk.gov.companieshouse.api.testdata.model.rest.RolesSpec;
+import uk.gov.companieshouse.api.testdata.model.rest.UsersSpec;
+import uk.gov.companieshouse.api.testdata.model.rest.UserTestData;
+import uk.gov.companieshouse.api.testdata.repository.RoleRepository;
+import uk.gov.companieshouse.api.testdata.repository.UserRepository;
+import uk.gov.companieshouse.api.testdata.service.UserService;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+import java.time.LocalDate;
+import java.time.ZoneId;
+import java.util.*;
+
+@Service
+public class UserServiceImpl implements UserService {
+    private static final ZoneId ZONE_ID_UTC = ZoneId.of("UTC");
+    private static final Logger LOG = LoggerFactory.getLogger(Application.APPLICATION_NAME);
+
+    @Autowired
+    private UserRepository repository;
+
+    @Autowired
+    private RoleRepository roleRepository;
+
+    @Override
+    public boolean userExists(String userId) {
+        Optional<Users> existingUser = repository.findById(userId);
+        return existingUser.isPresent();
+    }
+
+    @Override
+    public UserTestData create(UsersSpec usersSpec) throws DataException {
+        var dateNow = LocalDate.now().atStartOfDay(ZONE_ID_UTC).toInstant();
+        long timestamp = System.currentTimeMillis();
+        final String password = usersSpec.getPassword();
+        final var user = new Users();
+        List<RolesSpec> roleList = usersSpec.getRoles();
+        if(roleList!=null){
+            var role = new Roles();
+            List<String> rolesList = new ArrayList<>();
+            for (var roleData : roleList) {
+                if (roleData.getId() == null || roleData.getPermissions() == null) {
+                    throw new DataException("Role does not exist");
+                }
+                role.setId(roleData.getId()+"-playwright-role"+timestamp);
+                role.setPermissions(roleData.getPermissions());
+                rolesList.add(role.getId());
+                roleRepository.save(role);
```

Comment:
Will this not create roles even if one of them failed? 

Say we have 3 specs and the third one is wrong, it will create the other 2 before failing. Maybe we should not allow that? Worth adding unit test for this scenario too

Reply:
changed the logic and added unit tests as well